sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -1307,18 +1309,37 @@ fn check_param_type<Mode: ExecutionMode>(
                 Mode::allow_arbitrary_values() || !abilities.has_key(),
                 "Raw value should never be an object"
             );
-            ty
+                if ty != param_ty {
+                    return Err(command_argument_error(
+                            CommandArgumentError::TypeMismatch,
+                            idx,
+                    ))
+                }
+        }
+        Value::Object(obj) =>  {
+            let ty = &obj.type_;
+            if ty != param_ty {
+                return Err(command_argument_error(
+                        CommandArgumentError::TypeMismatch,
+                        idx,
+                ))
+            }
+        }
+        Value::Receiving(_, _) =>  {
+            if let Type::StructInstantiation(idx, targs) = param_ty {
+                let Some(s) = context
+                    .session
+                    .get_struct_type(*idx) else {
+                        invariant_violation!("sui::transfer::Receiving struct not found in session")
+                    };
+                let resolved_struct = get_struct_ident(&s);
+                if resolved_struct == RESOLVED_RECEIVING_STRUCT && targs.len() == 1 {
+                    return Ok(())
+                }
+            }
```

Comment:
else Error? I think?
(should add a test for that too if it's missing)

Reply:
Good catch! Need to write a test for this + I want to write some more around this area too now :)

Reply:
Fixed + wrote tests to exercise this case + wrote a bunch of other tests